services/user.py

```diff
@@ -0,0 +1,57 @@
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> User:
+    user = User()
```

Comment:
Before creating a new user, check if a user with the same username already exists to avoid integrity errors. Consider using `User.objects.filter(username=username).exists()` to perform this check.